app/controllers/KairosDbProxyController.java

```diff
@@ -173,6 +187,29 @@ public KairosDbProxyController(
         }
     }
 
+    MetricsQuery addMergeAggregator(final MetricsQuery metricsQuery) {
+        final List<Metric> newMetrics = new ArrayList<>();
+        for (Metric metric : metricsQuery.getMetrics()) {
+            List<Aggregator> newAggregators = new ArrayList<>();
+            if (!metric.getAggregators().isEmpty() && !(metric.getAggregators().get(0).getName()).equals("merge")
+                    && metric.getAggregators().stream().noneMatch(
+                            aggregator -> NONHISTAGGREGATORS.contains(aggregator.getName()))) {
+                final Optional<Sampling> originalSampling = metric.getAggregators().get(0).getSampling();
```

Comment:
Hmm, I think we'll want to copy the sampling from the first aggregator _that specifies a sampling_, rather than just the first aggregator.

Example where this is important: I, a desktop developer, have measured some time in milliseconds, but I want to display it in seconds on Grafana. So I query my metric and use the aggregators
```
[{"name": "div", "divisor": 1000},
 {"name": "percentile", "percentile": 0.75, "sampling": {"value": 1, "unit": "HOURS"}}
]
```
As written, the code will set the merge aggregator's `sampling` to `null`, but we actually want "one hour."

Additionally, we'll want to copy over the `align_sampling` and `align_start_time` fields from that first-aggregator-with-sampling. It's kinda hard to describe what they do ([docs](https://kairosdb.github.io/docs/build/html/restapi/QueryMetrics.html#metric-properties)), but they're conceptually part of `sampling`, and should get copied over along with that field.

Reply:
👍